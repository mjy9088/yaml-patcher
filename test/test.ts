import { exit } from 'process';
import { readFile, writeFile } from 'fs/promises';
import yaml from 'js-yaml';
import YamlPatcher from '../src/index.js';

interface Content extends Record<string, unknown> {
  date?: string;
  lastmod?: string;
  tags?: string[];
}

(async function updateTest(): Promise<void> {
  const old = (await readFile('test/before.yml')).toString();
  const expected = (await readFile('test/after.yml')).toString();
  const before = yaml.load(old) as Content;
  const after = {
    date: new Date().toISOString(),
    ...before,
    lastmod: '2022-02-09T12:34:56.789Z',
    tags: ['before', ...(before.tags || []), 'after'],
    newProperty: 'added',
    custom: 'Modified ' + before.custom + '!',
  };
  const result = YamlPatcher(old, after);
  if (result !== expected) {
    console.error("Failed");
    exit(1);
  }
})();
