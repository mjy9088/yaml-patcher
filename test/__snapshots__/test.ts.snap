// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Object {
  "custom": "Modified test!",
  "date": "2022-02-08T17:34:18.844Z",
  "lastmod": "2022-02-09T12:34:56.789Z",
  "newProperty": "added",
  "slug": "yaml-test",
  "tags": Array [
    "before",
    "Hello",
    "world!",
    "after",
  ],
  "title": "YAML Test",
}
`;
